front/src/utils/hooks/OsrdAuth.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { getIsUserLogged, getUsername } from 'reducers/user/userSelectors';
+import { osrdGatewayApi } from 'common/api/osrdGatewayApi';
+
+type AuthHookData = {
+  username?: string;
+  isUserLogged: boolean;
+  isLocalBackend: boolean;
+  isLoading: boolean;
+  logout: () => void;
+};
+
+function useAuth(): AuthHookData {
+  const isUserLogged = useSelector(getIsUserLogged);
+  const username = useSelector(getUsername);
+
+  const [login, { isLoading: isAuthenticateLoading }] =
+    osrdGatewayApi.endpoints.login.useMutation();
+  const [logout] = osrdGatewayApi.endpoints.logout.useMutation();
+
+  useEffect(() => {
+    if (!isUserLogged && !isAuthenticateLoading) {
+      login();
+    }
+  }, [isUserLogged]);
+
+  return {
+    username,
+    isUserLogged,
+    isLocalBackend: false,
```

Comment:
Do you need this ? I think you don't need it anymore, as it is always false. You can also remove its usage in App.jsx